manta-trusted-setup/src/groth16/ppot/serialization.rs

```diff
@@ -0,0 +1,964 @@
+        let mut g2_deser = Vec::<G2Affine>::new();
+
+        println!("Deserializing now ");
+
+        for i in 0..N {
+            let start = i * <PpotCeremony as Serializer<G2Affine, G2>>::uncompressed_size(&g2[0]);
+            let end =
+                (i + 1) * <PpotCeremony as Serializer<G2Affine, G2>>::uncompressed_size(&g2[0]);
+            let mut temp = &file[start..end];
+            match <PpotCeremony as Deserializer<G2Affine, G2>>::deserialize_uncompressed(&mut temp)
+            {
+                Ok(point) => g2_deser.push(point),
+                Err(e) => {
+                    println!("Error {:?} occurred on point {:?}", e, i);
+                }
+            }
+        }
+        assert_eq!(g2, g2_deser);
+    }
+
+    /// Compares the accumulators stored in response_0071 and challenge_0072
+    #[ignore] // NOTE: Adds `ignore` such that CI does NOT run this test while still allowing developers to test.
+    #[test]
+    pub fn compare_response_challenge_accumulators_test() {
+        use crate::groth16::ppot::kzg::PerpetualPowersOfTauCeremony;
+        use ark_std::{fs::OpenOptions, time::Instant};
+        use memmap::MmapOptions;
+
+        const POWERS: usize = 1 << 5;
+        /// Configuration for a Phase1 Ceremony large enough to support MantaPay circuits
+        pub type SubCeremony = PerpetualPowersOfTauCeremony<PpotSerializer, POWERS>;
+
+        // Try to load `./challenge` from disk.
+        println!("Reading accumulator from challenge file");
+        let now = Instant::now();
+        let reader = OpenOptions::new()
+            .read(true)
+            .open("/Users/thomascnorton/Documents/Manta/trusted-setup/challenge_0072")
+            .expect("unable open `./challenge` in this directory");
+        // Make a memory map
+        let challenge_map = unsafe {
+            MmapOptions::new()
+                .map(&reader)
+                .expect("unable to create a memory map for input")
+        };
+        let challenge_acc =
+            read_subaccumulator::<SubCeremony>(&challenge_map, Compressed::No).unwrap();
+        println!("Read uncompressed accumulator in {:?}", now.elapsed());
+
+        // Try to load `./response` from disk.
+        println!("Reading accumulator from response file");
+        let now = Instant::now();
+        let reader = OpenOptions::new()
+            .read(true)
+            .open("/Users/thomascnorton/Documents/Manta/trusted-setup/response_0071")
+            .expect("unable open `./response` in this directory");
+        // Make a memory map
+        let response_map = unsafe {
+            MmapOptions::new()
+                .map(&reader)
+                .expect("unable to create a memory map for input")
+        };
+        let response_acc =
+            read_subaccumulator::<SubCeremony>(&response_map, Compressed::Yes).unwrap();
+        println!("Read compressed accumulator in {:?}", now.elapsed());
+
+        assert_eq!(challenge_acc, response_acc)
+    }
+}
```

Comment:
Would it make sense to also run some tests against known (de)serialization implementations?

Reply:
Maybe so, and besides this test is really inelegantly written, I used a lot of copy-paste